core/src/test/java/org/elasticsearch/search/SearchModuleTests.java

```diff
@@ -99,16 +99,16 @@ public void testRegisteredQueries() throws IOException {
         IndicesQueriesRegistry indicesQueriesRegistry = module.getQueryParserRegistry();
         XContentParser dummyParser = XContentHelper.createParser(new BytesArray("{}"));
         for (String queryName : supportedQueries) {
-            indicesQueriesRegistry.lookup(queryName, dummyParser, ParseFieldMatcher.EMPTY);
+            indicesQueriesRegistry.lookup(queryName, ParseFieldMatcher.EMPTY, dummyParser::getTokenLocation);
```

Comment:
> My understanding is that this still allocates.

ok you didn't mean allocating the new XContentLocation as part of calling getTokenLocation but the fact that we still  allocate an object, the Supplier rather than XContentLocation?